mpactpy/pin.py

```diff
@@ -182,6 +182,30 @@ def axial_merge(self, pin: Pin) -> Pin:
         return Pin(pinmesh_template, materials)
 
 
+    def with_height(self, height: float) -> Pin:
+        """ Method for changing the height of 2D Pin
+
+        Parameters
+        ----------
+        height : float
+            The height of the new 2D pin
+
+        Returns
+        -------
+        Pin
+            The new pin with the new height
+        """
+
+        pinmesh = deepcopy(self.pinmesh)
+
+        assert len(pinmesh.zvals), \
```

Comment:
Do you mean that len(pinmesh.zvals) should be 0 if the pin is 2D?  As it is, it's seems to be asserting the opposite. 
```suggestion
        assert len(pinmesh.zvals) == 0, \
```
Or possibly:
```suggestion
        assert len(pinmesh.zvals) == 1, \
```
If only 1 zval is expected.

Reply:
Yeah, that should have been `== 1`